.github/workflows/run-tests.yml

```diff
@@ -254,7 +254,7 @@ jobs:
           cache: 'maven'
 
       - name: Set up JDK 11 # get rid of this after https://github.com/snowflakedb/snowflake-jdbc/issues/589 is fixed
-        if: ${{ matrix.testSystem == 'snowflake' }}
+        if: ${{ github.event_name == 'push' && github.ref == 'refs/heads/master' && matrix.testSystem == 'snowflake' }}
```

Comment:
Should you leave the `github.event_name == 'push'` here and on the below two steps?

Reply:
I still think it should be `pull_request` all the way.